components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementService.java

```diff
@@ -65,15 +66,33 @@ public abstract void createApplication(ServiceProvider serviceProvider, String t
      * Creates a service provider with basic information and returns the created service provider. First we need to
      * create an internal role with the application name. Only the users in this role will be able to edit/update
      * the application.Then the user will assigned to the created role.
+     *
      * @param serviceProvider Service Provider Name
      * @param tenantDomain Tenant Domain
      * @param username User Name
      * @return created service provider
      * @throws IdentityApplicationManagementException
      */
+    @Deprecated
     public abstract ServiceProvider addApplication(ServiceProvider serviceProvider, String tenantDomain, String
             username) throws IdentityApplicationManagementException;
 
+    /**
+     * Creates a service provider with basic information and returns the created service provider. First we need to
+     * create an internal role with the application name. Only the users in this role will be able to edit/update
+     * the application.Then the user will assigned to the created role.
+     *
+     * @param serviceProvider Service Provider Name
+     * @param tenantDomain Tenant Domain
+     * @param username User Name
+     * @param templateName SP template name
+     * @return created service provider
+     * @throws IdentityApplicationManagementException
+     */
+    public abstract ServiceProvider addApplication(ServiceProvider serviceProvider, String tenantDomain,
```

Comment:
What's the duty of this method?

Reply:
This will be creating a service provider, using the template provided. Changed this method name to "createApplicationWithTemplate".